torch/CMakeLists.txt

```diff
@@ -465,3 +465,129 @@ install(FILES
+  ${TORCH_SRC_DIR}/csrc/Exceptions.cpp
+  ${TORCH_SRC_DIR}/csrc/TypeInfo.cpp
+  ${TORCH_SRC_DIR}/csrc/Generator.cpp
+  ${TORCH_SRC_DIR}/csrc/Layout.cpp
+  ${TORCH_SRC_DIR}/csrc/Module.cpp
+  ${TORCH_SRC_DIR}/csrc/PtrWrapper.cpp
+  ${TORCH_SRC_DIR}/csrc/Size.cpp
+  ${TORCH_SRC_DIR}/csrc/Storage.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/functions/init.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/generated/python_functions.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/generated/python_nn_functions.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/generated/python_torch_functions.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/generated/python_variable_methods.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/init.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/python_anomaly_mode.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/python_cpp_function.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/python_engine.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/python_function.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/python_hook.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/python_legacy_variable.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/python_variable.cpp
+  ${TORCH_SRC_DIR}/csrc/autograd/python_variable_indexing.cpp
+  ${TORCH_SRC_DIR}/csrc/byte_order.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/batched/BatchTensor.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/init.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/passes/onnx.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/passes/onnx/fixup_onnx_loop.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/passes/onnx/prepare_division_for_onnx.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/passes/onnx/peephole.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/passes/to_batch.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/python_arg_flatten.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/python_interpreter.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/python_ir.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/python_tracer.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/script/init.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/script/lexer.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/script/module.cpp
+  ${TORCH_SRC_DIR}/csrc/jit/script/python_tree_views.cpp
+  ${TORCH_SRC_DIR}/csrc/nn/THNN.cpp
+  ${TORCH_SRC_DIR}/csrc/onnx/init.cpp
+  ${TORCH_SRC_DIR}/csrc/serialization.cpp
+  ${TORCH_SRC_DIR}/csrc/tensor/python_tensor.cpp
+  ${TORCH_SRC_DIR}/csrc/utils.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/cuda_lazy_init.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/invalid_arguments.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/object_ptr.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/python_arg_parser.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/tensor_apply.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/tensor_conversion_dispatch.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/tensor_dtypes.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/tensor_flatten.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/tensor_layouts.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/tensor_list.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/tensor_new.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/tensor_numpy.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/tensor_types.cpp
+  ${TORCH_SRC_DIR}/csrc/utils/tuple_parser.cpp
+  )
+
+set(TORCH_PYTHON_INCLUDE_DIRECTORIES
+  ${TORCH_ROOT}
+  ${TORCH_ROOT}/aten/src
+  ${TORCH_ROOT}/aten/src/TH
+
+  ${CMAKE_BINARY_DIR}
```

Comment:
you should declare a project dependency on Caffe2, which will auto-include these dirs in the Torch project.
Go by Orion's desk for cmake 101 :)